### PR TITLE
FEATURE: ONE-4276 Auto resize behavior in Pivot table SDK component

### PR DESCRIPTION
**Pull request by vlastikcz**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2734/files
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2518

# Related Jira tasks

### CODE REVIEW

**Comment by vlastikcz** [comment]
extended test - storybook

**Comment by pbenes** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6065/ : FAILURE in 8m 46s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1692/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3134/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1693/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2513/ : UNSTABLE in 17m 33s

**Comment by vlastikcz** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by vlastikcz** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6067/ : FAILURE in 9m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1697/ : SUCCESS in 10s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3136/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1698/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2516/ : UNSTABLE in 17m 31s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by pbenes** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6072/ : FAILURE in 7m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1703/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3138/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1704/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2518/ : UNSTABLE in 18m 24s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6074/ : SUCCESS in 9m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1707/ : SUCCESS in 4s

**Comment by ivan-nejezchleb** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1384.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1708/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2520/ : FAILURE in 28m 08s

**Comment by pbenes** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6075/ : SUCCESS in 9m 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1709/ : SUCCESS in 8s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3140/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1710/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2521/ : UNSTABLE in 14m 46s

**Comment by pbenes** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6081/ : SUCCESS in 8m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1718/ : SUCCESS in 6s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3146/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1719/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2526/ : UNSTABLE in 14m 46s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by pbenes** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6085/ : SUCCESS in 4m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1723/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3149/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1724/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2528/ : SUCCESS in 14m 28s

**Comment by pbenes** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1384.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1727/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2530/ : FAILURE in 23m 39s

File: src/components/core/PivotTable.tsx [hunk 1]
```diff
@@ -243,6 +263,15 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
             if (agGridDataSourceUpdateNeeded) {
                 this.updateAGGridDataSource();
             }
+
+            const dataSourceChanged =
+                this.props.dataSource.getFingerprint() !== prevProps.dataSource.getFingerprint();
+            if (dataSourceChanged || totalsPropsChanged || totalsStateChanged) {
+                this.columnWidths = {};
```
Comment (ivanmjartan) [inline src/components/core/PivotTable.tsx]: just wonder why columnWidth is not in state ? thats because double render?

Comment (pbenes) [reply src/components/core/PivotTable.tsx]: yes, that's the case

File: src/components/core/tests/PivotTable.spec.tsx [hunk 2]
```diff
@@ -27,6 +26,7 @@ import { waitFor } from "../../filters/AttributeFilter/tests/utils";
 import { IGridCellEvent } from "../pivotTable/agGridTypes";
 import { IDrillEventExtended } from "../../../interfaces/DrillEvents";
 import { Execution } from "@gooddata/typings";
+import noop = require("lodash/noop");
```
Comment (patrikbraborec) [inline src/components/core/tests/PivotTable.spec.tsx]: Is it correct import? Cannot we use `import {noop} from "lodash";`?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6090/ : SUCCESS in 22m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1731/ : SUCCESS in 10s

File: src/components/core/tests/PivotTable.spec.tsx (hunk 2 above)
Comment (pbenes) [reply src/components/core/tests/PivotTable.spec.tsx]: We do it this way on other places to avoid full lodash import

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6094/ : SUCCESS in 13m 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1735/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6113/ : SUCCESS in 5m 32s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1758/ : SUCCESS in 5s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6157/ : SUCCESS in 17m 38s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1812/ : SUCCESS in 4s

**Comment by pbenes** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3185/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1813/ : SUCCESS in 31s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2570/ : SUCCESS in 14m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6163/ : SUCCESS in 13m 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1818/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3188/ : FAILURE in 14m 50s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6164/ : SUCCESS in 5m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1502/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2111/ : FAILURE in 34m 46s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6165/ : SUCCESS in 8m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1820/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3190/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1821/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2573/ : SUCCESS in 17m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6173/ : SUCCESS in 5m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1833/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3196/ : SUCCESS in 23m 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6180/ : SUCCESS in 11m 28s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1505/ : SUCCESS in 51s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2117/ : SUCCESS in 23m 37s

**Pull request merged**